apps/src/applab/applab.js

```diff
@@ -1196,20 +1178,18 @@ Applab.execute = function () {
     }
   }
 
-  if (isMakerEnabled(getStore().getState())) {
-    connectToMakerBoard()
-        .then(board => {
-          board.installOnInterpreter(codegen, Applab.JSInterpreter);
-          makerCommands.injectBoardController(board);
-          board.once('disconnect', () => studioApp.resetButtonClick());
-          makerBoard = board;
-        })
-        .catch(error => studioApp.displayPlayspaceAlert('error',
-            <div>{`Board connection error: ${error}`}</div>))
-        .then(Applab.beginVisualizationRun);
-  } else {
-    Applab.beginVisualizationRun();
-  }
+    makerToolkit.connect({
+      interpreter: Applab.JSInterpreter,
+      onDisconnect: () => studioApp.resetButtonClick(),
+    })
+        .then(Applab.beginVisualizationRun)
+        .catch(error => {
+          // Don't just throw any error away, but squelch errors that we already
+          // handle gracefully (like early disconnect or a missing board).
+          if (!(error instanceof makerToolkit.MakerError)) {
+            Applab.log(error);
```

Comment:
I tested this out and `Applab.log` puts the error in the student's debug console (which I figured might be helpful to us in screenshots/bug reports) but also drops the error object into the browser console where it's inspectable.

A benefit of letting the error bubble all the way up would be New Relic noticing and logging the event, which could be helpful.

However, rethrowing may be difficult here; we're inside a promise chain, so I believe throwing here would result in an "Unhandled promise rejection" error that might undermine the benefit of New Relic in this case.

There's a hacky workaround for this:
```javascript
.catch(error => setTimeout(() => throw error, 0))
```
...but I don't feel great about it.  I'll go ahead and try an unhandled promise rejection and see what sort of information gets included.